tests/cases_compute_probabilistic_metrics.py

```diff
@@ -0,0 +1,821 @@
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+from scipy.stats import norm
+
+from gval.utils.schemas import Prob_metrics_df
+from gval.utils.loading_datasets import _create_xarray_pairs
+
+nodata_value = np.nan
+encoded_nodata_value = -9999
+upper_left = (-95.2, 37.2)
+lower_right = (-94.8, 36.8)
+shapes = "circle"
+return_datasets = False
+sizes = 50
+band_dim_name = "member"
+
+
+def case_compute_prob_metrics_brier_score_success():
+    expected_xr = xr.DataArray(0.22347608, coords={"spatial_ref": 0})
+
+    # compute_kwargs
+    compute_kwargs = {
+        "metric_kwargs": {"brier_score": {"member_dim": "member", "keep_attrs": True}},
+        "return_on_error": "error",
+    }
+
+    # creating expected df. expected_xr is added later due to pd dtype issues
+    expected_df = Prob_metrics_df(
+        pd.DataFrame({"band": ["member"], "brier_score": [None]})
+    )
+    expected_df.loc[0, "brier_score"] = expected_xr
+
+    # Band parameters for candidate
+    # background value, circle value, circle center, and circle radius
+    band_params_candidate = [
+        (0, 1, (7, 7), 3),  # Band 1
+        (0, 0, (12, 12), 5),  # Band 2
+        (1, 1, (17, 17), 7),  # Band 3
+    ]
+
+    # Band parameters for benchmark
+    # background value, circle value, circle center, and circle radius
+    band_params_benchmark = [
+        (0, 1, (8, 8), 4),  # Band 1
+        (0, 0, (13, 13), 6),  # Band 2
+        (1, 1, (18, 18), 8),  # Band 3
+    ]
+
+    # generate xarray pairs
+    candidate_map, benchmark_map = _create_xarray_pairs(
+        upper_left,
+        lower_right,
+        sizes,
+        band_params_candidate,
+        band_params_benchmark,
+        nodata_value,
+        encoded_nodata_value,
+        shapes,
+        band_dim_name,
+        return_datasets,
+    )
+
+    return candidate_map, benchmark_map, compute_kwargs, expected_df
+
+
+def case_compute_prob_metrics_crps_ensemble_success():
+    return_datasets = False
+    sizes = (50, 60)
+    expected_xr = xr.DataArray(44.65896392, coords={"spatial_ref": 0})
+
+    # compute_kwargs
+    compute_kwargs = {
+        "metric_kwargs": {
+            "crps_ensemble": {"member_dim": "member", "keep_attrs": True}
+        },
+        "return_on_error": None,
+    }
+
+    # creating expected df. expected_xr is added later due to pd dtype issues
+    expected_df = Prob_metrics_df(
+        pd.DataFrame({"band": ["member"], "crps_ensemble": [None]})
+    )
+    expected_df.loc[0, "crps_ensemble"] = expected_xr
+
+    # Band parameters for candidate
+    # background value, circle value, circle center, and circle radius
+    band_params_candidate = [
+        (100, 300, (20, 20), 7),
+        (150, 350, (30, 30), 10),
+        (200, 400, (40, 40), 8),
+    ]
+
+    # Band parameters for benchmark
+    # background value, circle value, circle center, and circle radius
+    band_params_benchmark = [
+        (145, 360, (25, 25), 15),
+        # (160, 360, (2, 2), 3),
```

Comment:
Unless there is a purpose remove commented lines.

Reply:
Removed.